hack/conformance-test.sh

```diff
@@ -18,22 +18,65 @@
 # against an already-setup cluster for which there is no automated
 # setup, teardown, and other cluster/... scripts.
 #
-# The user must export these environment variables:
-# KUBE_MASTER_IP to the ip address of the master.
-# AUTH_CONFIG to the argument of the "--auth_config=" flag.
-# If certs required, set CERT_DIR.
-# 
-# Example to test against a local vagrant cluster:
-# declare -x AUTH_CONFIG="$HOME/.kubernetes_vagrant_auth"
-# declare -x KUBE_MASTER_IP=10.245.1.2
-# hack/conformance-test.sh
-if [[ -z "KUBE_MASTER_IP" ]]; then
-  echo "Must set KUBE_MASTER_IP before running conformance test."
-  exit 1
-fi
-if [[ -z "AUTH_CONFIG" ]]; then
-  echo "Must set AUTH_CONFIG before running conformance test."
-  exit 1
-fi
+# The user must:
+#  - Setup a Kubernetes cluster with $NUM_MINIONS nodes.
+#  - Provide a Kubeconfig file whose current context is set to the
+#    cluster to be tested, and with suitable auth setting.
+#  - Specify the location of that kubeconfig with, e.g.:
+#    declare -x KUBECONFIG="$HOME/.kube/config"
+#  - Specify the location of the master with, e.g.:
+#    declare -x KUBE_MASTER_IP="1.2.3.4"
+#  - Make sure only essential pods are running and there are no failed/pending pods.
+# Then run:
+#    hack/conformance-test.sh 2>&1 | tee conformance.$(date +%FT%T%z).log
+
+: ${KUBECONFIG:?"Must set KUBECONFIG before running conformance test."}
+: ${KUBE_MASTER_IP:?"Must set KUBE_MASTER_IP before running conformance test."}
+echo "Conformance test using $KUBECONFIG against master at $KUBE_MASTER_IP"
+echo -n "Conformance test run date:"
+date
+echo -n "Conformance test SHA:"
+HEAD_SHA=$(git rev-parse HEAD)
+echo $HEAD_SHA
+echo "Conformance test version tag(s):"
+git show-ref | grep $HEAD_SHA | grep refs/tags
+echo
+
+# The Conformance Test ensures that different implementations of Kubernetes clusters
+# have compatible "user" APIs.  It does not test:
+# - node management
+# - optional features
+# - implementation specific things
+# - performance, scale, or reliability
+# - volume types that may not be available on all implementations.
+# - known flaky tests.
+#
+# It runs a whitelist of tests.  This whitelist was assembled at commit
+# b70b7084c93d4ce80b7463f48c23d5ac04edb2b1 starting from this list of tests:
+#   grep -h 'It(\|Describe(' -R test
+#
+# List of test name patterns not included and why not included:
```

Comment:
listing the test patterns *excluded* but then using a regex to only *include* certain tests basically guarantees that at least one of the lists will become out of date. are we OK with this?